module/zfs/spa_misc.c

```diff
@@ -555,9 +584,14 @@ spa_config_held(spa_t *spa, int locks, krw_t rw)
 		spa_config_lock_t *scl = &spa->spa_config_lock[i];
 		if (!(locks & (1 << i)))
 			continue;
+#ifdef ZFS_DEBUG
 		if ((rw == RW_READER &&
 		    !zfs_refcount_is_zero(&scl->scl_count)) ||
 		    (rw == RW_WRITER && scl->scl_writer == curthread))
+#else
+		if ((rw == RW_READER && scl->scl_count != 0) ||
```

Comment:
It looks like the scl_lock isn't held here.  I guess this is OK since this function is inherently racy - the answer could change as soon as we return.  So it's only used for assertions.